invenio_rdm_records/marshmallow/json.py

```diff
@@ -433,6 +433,7 @@ class MetadataSchemaV1(BaseSchema):
                                attribute="embargo_date")
     _community = Nested(CommunitySchemaV1, data_key="community",
                         attribute="community")
+    _communities = fields.Raw(data_key="_communities", attribute="_communities")
```

Comment:
[Discussion] Can this new field replace `_community` or is it complementary? I think it's complementary because it needs additional functionality before it can account for a record that was added and then removed just based on the elasticsearch mapping above. But I would like it to replace, because it would simplifiy communities by making all the community stuff go in one field (which can then be sub-divided).

Here is the structure I see:
 
`_communities` -> `_communities_statuses`

```python
class CommunitiesRequestV1(BaseSchema):
    """Community Request Schema."""

    id = SanitizedUnicode(required=True)  # difference between id and request_id ?
    community_id = SanitizedUnicode(required=True)  # comid
    title = SanitizedUnicode(required=True)
    request_id = SanitizedUnicode(required=True)
    created_by = fields.Integer(required=True)


class CommunityStatusV1(BaseSchema):
    """Status of a community request."""
    pending = fields.List(Nested(CommunitiesRequestV1))
    accepted = fields.List(Nested(CommunitiesRequestV1))
    rejected = fields.List(Nested(CommunitiesRequestV1))


class MetadataSchemaV1(BaseSchema):
    # ....
    _communities_statuses = Nested(CommunityStatusV1)
```

If that is the meaning behind these mapping objects.

Reply:
I agree that it should be eventually fleshed out into a proper field with schemas, etc. Keep in mind that we also don't dump/store this information in the record's JSON and that's the reason there is no JSONSchema addition in this PR (more on that later down).

Its purpose for now in the integration is to serialize the [ES indexed value](https://github.com/inveniosoftware/invenio-communities/blob/17059f7f10ed5da6b21c0cfa6b2e2c8c83e20656/invenio_communities/records/indexer.py#L15) in the `/api/records` search response so that it can be used in the [curation page](https://zenodo-rdm.web.cern.ch/communities/foo/curate). Actually for completeness, I'll add a `Method` field that dumps this dictionary into the `_communities` field so that the item resource (`/api/records/<recid>`) also has the field serialized.

The basic idea for how we proceed with serialization is that the `RecordCommunitiesMixin` (added to the API Record class) adds a `record.communities` property (similar to `record.files`). You can then as part of `record.dumps()` do a `record.communities.dumps()` (or [`record.communities.as_dict()`](https://github.com/inveniosoftware/invenio-communities/blob/17059f7f10ed5da6b21c0cfa6b2e2c8c83e20656/invenio_communities/records/api.py#L351)) call to get the data that will then become the input to the Marshmallow serializer.